src/main_thread/decrypt/content_decryptor.ts

```diff
@@ -679,6 +693,19 @@ export default class ContentDecryptor extends EventEmitter<IContentDecryptorEven
     mediaKeysData: IAttachedMediaKeysData,
   ): boolean {
     const { stores, options } = mediaKeysData;
+    /**
+     * On Safari using Directfile, the old EME implementation triggers
+     * the "webkitneedkey" event instead of "encrypted". There's an issue in Safari
+     * where "webkitneedkey" fires too early before all tracks are added from an HLS playlist.
+     * Safari incorrectly assumes some keys are missing for these tracks,
+     * leading to repeated "webkitneedkey" events. Because RxPlayer recognizes
+     * it already has a session for these keys and ignores the events,
+     * the content remains frozen. To resolve this, we handle the event
+     * assuming no session exists for the key.
+     */
+    if (this._context.isDirectFile && isSafariDesktop) {
+      return false;
+    }
```

Comment:
There are some issues I have with this solution:
 
  1. Though from the comment, it appears that we want to provide a work-around for specifically the `webkitneedkey` event, it' s actually in a `_tryToUseAlreadyCreatedSession` private function which has no knowledge of that event and cannot guarantee that the data was sent from this event.

  2.   What happens of the previous "session" here (if there's one on Safari, I don't remember the webkit-prefixed EME API perfectly)? Is it just dangling at the browser-level? And if it's resources seem properly disposed, is it properly removed from the `LoadedSessionsStore` (our in-JS session cache) or is it leaking there?